lib/eex/lib/eex/compiler.ex

```diff
@@ -34,67 +42,89 @@ defmodule EEx.Compiler do
-                      %{state | quoted: [], line: line, start_line: start_line})
+      generate_buffer(rest, state.engine.handle_begin(buffer), [contents | scope], %{
+        state | quoted: [], line: line, start_line: start_line
+      })
+
     buffer = state.engine.handle_expr(buffer, IO.chardata_to_string(mark), contents)
     generate_buffer(rest, buffer, scope, state)
   end
 
   defp generate_buffer([{:middle_expr, line, '', chars} | rest], buffer, [current | scope], state) do
     {wrapped, state} = wrap_expr(current, line, buffer, chars, state)
-    generate_buffer(rest, state.engine.handle_begin(buffer), [wrapped | scope], %{state | line: line})
+
+    generate_buffer(rest, state.engine.handle_begin(buffer), [wrapped | scope], %{
+      state | line: line
+    })
   end
 
-  defp generate_buffer([{:middle_expr, line, modifier, chars} | t], buffer, [_ | _] = scope, state) do
-    message = "unexpected beginning of EEx tag \"<%#{modifier}\" on \"<%#{modifier}#{chars}%>\", " <>
-              "please remove \"#{modifier}\" accordingly"
-    :elixir_errors.warn line, state.file, message
+  defp generate_buffer(
+         [{:middle_expr, line, modifier, chars} | t],
+         buffer,
+         [_ | _] = scope,
+         state
+       ) do
+    message =
+      "unexpected beginning of EEx tag \"<%#{modifier}\" on \"<%#{modifier}#{chars}%>\", " <>
+        "please remove \"#{modifier}\" accordingly"
+
+    :elixir_errors.warn(line, state.file, message)
     generate_buffer([{:middle_expr, line, '', chars} | t], buffer, scope, state)
     # TODO: Make this an error on Elixir v2.0 since it accidentally worked previously.
     # raise EEx.SyntaxError, message: message, file: state.file, line: line
   end
 
   defp generate_buffer([{:middle_expr, line, _, chars} | _], _buffer, [], state) do
-    raise EEx.SyntaxError, message: "unexpected middle of expression <%#{chars}%>",
-                           file: state.file, line: line
+    raise EEx.SyntaxError,
+      message: "unexpected middle of expression <%#{chars}%>",
+      file: state.file,
+      line: line
   end
 
   defp generate_buffer([{:end_expr, line, '', chars} | rest], buffer, [current | _], state) do
     {wrapped, state} = wrap_expr(current, line, buffer, chars, state)
-    tuples = Code.string_to_quoted!(wrapped, [line: state.start_line, file: state.file])
+    tuples = Code.string_to_quoted!(wrapped, line: state.start_line, file: state.file)
     buffer = insert_quoted(tuples, state.quoted)
     {buffer, rest}
   end
 
   defp generate_buffer([{:end_expr, line, modifier, chars} | t], buffer, [_ | _] = scope, state) do
-    message = "unexpected beginning of EEx tag \"<%#{modifier}\" on end of expression \"<%#{modifier}#{chars}%>\", " <>
-              "please remove \"#{modifier}\" accordingly"
-    :elixir_errors.warn line, state.file, message
+    message =
+      "unexpected beginning of EEx tag \"<%#{modifier}\" on end of expression \"<%#{modifier}#{
```

Comment:
Please change this message so it uses concatenation in different places. For example:

```
"unexpected beginning of EEx tag \"<%#{modifier}\" on end of "  <>
  expression \"<%#{modifier}#{chars}%>\", please remove \"#{modifier}\" accordingly"
```

Or similar.